src/dag_combine.rs

```diff
@@ -0,0 +1,56 @@
+use digraph::PyDiGraph;
+use pyo3::prelude::*;
+use petgraph::algo;
+use pyo3::Python;
+use pyo3::types::PyAny;
+use std::cmp::Ordering;
+
+fn contains_weight(py: Python, dag: &PyDiGraph, obj: &PyObject) -> i32 {
+    let mut index :i32 = -1;
+
+    for node in dag.graph.node_indices() {
+        let weight = dag.graph.node_weight(node).unwrap();
+        if ! weight.is_none(py) {
+
+            let predicate_callable = |a: &PyAny, b: &PyAny| -> PyResult<bool> {
+                let res = a.compare(b)?;
+                Ok(res == Ordering::Equal)
+            };
+
+            if predicate_callable(obj.as_ref(py), weight.as_ref(py)).unwrap() {
+                index = node.index() as i32;
```

Comment:
Yeah, this is the potentially problematic piece I was alluding to above. This could cause an error if the index is larger than max positive i32 value.